lib/top.ex

```diff
@@ -0,0 +1,27 @@
+defmodule Crash.Top do
+  @moduledoc """
+  supervisor for Middleware
+  """
+  use Supervisor
+
+  def start_link(init_arg) do
+    Supervisor.start_link(__MODULE__, init_arg, name: __MODULE__)
+  end
+
+  @impl true
+  def init(_init_arg) do
+    IO.puts "Top init"
+    children = [
+      Supervisor.child_spec({Crash.Middleware, []}, id: :w1),
+      Supervisor.child_spec({Crash.Middleware, []}, id: :w2),
+      Supervisor.child_spec({Crash.Middleware, []}, id: :w3)
+    ]
```

Comment:
Привыкайте обходиться без хардкода везде, где это уместно.

```suggestion
  def start_link(init_arg \\ [:w1, :w2, :w3]) do
    Supervisor.start_link(__MODULE__, init_arg, name: __MODULE__)
  end

  @impl true
  def init(init_arg) do
    IO.puts "Top init"
    children =
      for worker <- init_arg do
        Supervisor.child_spec({Crash.Middleware, []}, id: worker)
      end
```